client/rest-high-level/src/main/java/org/elasticsearch/client/dataframe/transforms/DestConfig.java

```diff
@@ -72,11 +84,45 @@ public boolean equals(Object other) {
         }
 
         DestConfig that = (DestConfig) other;
-        return Objects.equals(index, that.index);
+        return Objects.equals(index, that.index) &&
+            Objects.equals(pipeline, that.pipeline);
     }
 
     @Override
     public int hashCode(){
-        return Objects.hash(index);
+        return Objects.hash(index, pipeline);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String index;
+        private String pipeline;
+
+        /**
+         * Sets which index to which to write the data
+         * @param index where to write the data
+         * @return The {@link Builder} with index set
+         */
+        public Builder setIndex(String index) {
+            this.index = index;
```

Comment:
Maybe `Objects.requireNonNull(index)` here so the error that the constructor will throw is traced to the root cause.